src/components/OcButton.spec.js

```diff
@@ -0,0 +1,146 @@
+  describe('disabled', () => {
+    let wrapper
+    beforeEach(() => {
+      wrapper = getWrapperWithProps({ disabled: true })
+    })
+    it('should have disabled attribute set to disabled', () => {
+      expect(wrapper.attributes("disabled")).toBe("disabled")
+    })
+    it("does not emit click event", async () => {
+      await wrapper.trigger("click")
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeFalsy()
+    })
+    it("should not route to provided router link", async () => {
+      wrapper = getWrapperWithPropsAndMockRouter({
+        type: "router-link",
+        to: {
+          name: 'some-route'
+        },
+        disabled: true
+      })
+      await wrapper.trigger("click")
+      await wrapper.vm.$nextTick()
+      expect(wrapper.emitted('click')).toBeFalsy()
+      console.log(wrapper.vm.$route.path)
+      expect(wrapper.vm.$route.path).toBe(route.path)
+    })
+  })
+  describe('different types of button', () => {
+    it.each`
+      type             | expectLink | expectButton | expectRouterLink
+      ${'a'}           | ${true}    | ${false}     | ${false}
+      ${'button'}      | ${false}   | ${true}      | ${false}
+      ${'router-link'} | ${false}   | ${false}     | ${true}
+    `
+    ('can behave as a $type', (
+      {type, expectLink, expectButton, expectRouterLink}) => {
+        const wrapper = getWrapperWithProps({ type: type })
+        expect(wrapper.find('a').exists()).toBe(expectLink)
+        expect(wrapper.find('button').exists()).toBe(expectButton)
+        expect(wrapper.find('router-link-stub').exists()).toBe(expectRouterLink)
+      })
+    it("can route to provided router link", async () => {
+      const wrapper = getWrapperWithPropsAndMockRouter({
+        type: "router-link",
+        to: {
+          name: 'some-route'
+        }
+      })
+      await wrapper.trigger("click")
+      await wrapper.vm.$nextTick()
+      expect(wrapper.vm.$route.path).toBe(route.path)
+    })
+  })
+  describe("different sizes of button", () => {
+    it.each`
+      size        | expectedClass
+      ${'small'}  | ${'oc-button-s'}
+      ${'medium'} | ${'oc-button-m'}
+      ${'large'}  | ${'oc-button-l'}
+      ${'x-small'}| ${'oc-button-undefined'}
+    `
+    ('when size prop is set as $size class $expectedClass should be assigned', (
+      {size, expectedClass}) => {
+        const wrapper = getWrapperWithProps({
+          size: size,
+        })
+        expect(wrapper.attributes("class")).toContain(expectedClass)
+      });
+  })
+  describe("default prop values", () => {
+    it.each`
+      name                 | expected
+      ${'size'}            | ${'oc-button-m'}
+      ${'variation'}       | ${'oc-button-passive'}
+      ${'justify content'} | ${'oc-button-justify-content-center'}
+      ${'gap size'}        | ${'oc-button-gap-m'}
+      ${'appearance'}      | ${'oc-button-passive-outline'}
+    `
+    ('should have $name as $expected', ({name, expected}) => {
```

Comment:
name is not used :thinking:

Reply:
name is used in the description

Reply:
👍

Reply:
I would have put it all in a normal `it`
with multiple `expect` in it, but OK - if you like that style its good ;-)